data_import/models.py

```diff
@@ -143,13 +135,20 @@ def generate_key(self, request):
         if request:
             # Log the entity that is requesting the key be generated
             new_key.ip_address = get_ip(request)
-            new_key.access_token = request.query_params.get("access_token", None)
-            if hasattr(request.auth, "application"):
-                # oauth2 project auth
-                new_key.project_id = request.auth.application.id
+            if hasattr(request, "query_params"):
+                new_key.access_token = request.query_params.get("access_token", None)
+            else:
+                new_key.access_token = None
+            if hasattr(request, "auth"):
+                if hasattr(request.auth, "application"):
+                    # oauth2 project auth
+                    new_key.project_id = request.auth.application.id
```

Comment:
This looks wrong, sorry I didn't catch this before.

The request.auth = an `OAuth2DataRequestProject`. This should have that project ID, which also works for the parent class `DataRequestProject`. (OnSite and OAuth2 child models share the same ID assignments, so an ID in either of those is also an ID in the parent model. Which means these are the universal way to look up any project.)

The application ID is related to the REST Framework and I don't expect it to match the IDs used by `DataRequestProject`, `OAuth2DataRequestProject` and `OnSiteDataRequestProject`. (And indeed, it doesn't.) This is going to result in mis-identifying the project that was used to download data, and we should probably simply wipe all logs before now because of the ambiguity.

Reply:
(the correct thing to do is the same as onsite: `request.auth.id`)